packages/engine-core/src/data-proxy/utils/request.ts

```diff
@@ -18,13 +19,20 @@ declare let fetch: typeof nodeFetch
  * @param options
  * @returns
  */
-export async function request(url: string, options: RequestOptions = {}): Promise<RequestResponse> {
+export async function request(
+  url: string,
+  options: RequestOptions & { clientVersion: string },
+): Promise<RequestResponse> {
   const jsRuntimeName = getJSRuntimeName()
 
-  if (jsRuntimeName === 'browser') {
-    return fetch(url, options)
-  } else {
-    return nodeFetch(url, options)
+  try {
+    if (jsRuntimeName === 'browser') {
+      return await fetch(url, options)
+    } else {
+      return await nodeFetch(url, options)
+    }
+  } catch (e) {
+    throw new NetworkError({ clientVersion: options.clientVersion })
```

Comment:
Would it make sense to:
1. Preserve original error message somehow?
2. Ensure that `e` is really related to network and not some other JS exception.